akka-actor/src/main/scala/akka/util/ByteString.scala

```diff
@@ -274,8 +322,43 @@ object ByteString {
     def asByteBuffers: scala.collection.immutable.Iterable[ByteBuffer] = bytestrings map { _.asByteBuffer }
 
     def decodeString(charset: String): String = compact.decodeString(charset)
+
+    private[akka] def writeToOutputStream(os: ObjectOutputStream): Unit = {
+      os.writeInt(bytestrings.length)
+      bytestrings.foreach(_.writeToOutputStream(os))
+    }
+
+    protected def writeReplace(): AnyRef = new SerializationProxy(this)
+  }
+
+  @SerialVersionUID(1L)
+  private class SerializationProxy(@transient private var orig: ByteString) extends Serializable {
+    private def writeObject(out: ObjectOutputStream) {
+      out.writeByte(orig.byteStringCompanion.SerializationIdentity)
+      orig.writeToOutputStream(out)
+    }
+
+    private def readObject(in: ObjectInputStream) {
+      val serializationId = in.readByte()
+
+      orig = Companion(from = serializationId).readFromInputStream(in)
+    }
+
+    private def readResolve(): AnyRef = orig
+  }
+
+  private[akka] object Companion {
+    private val companionMap = Seq(ByteString1, ByteString1C, ByteStrings).
```

Comment:
How about submitting a PR with that improvement @colinrgodsey? Would be very welcome :-)